hail/src/main/scala/is/hail/asm4s/joinpoint/JoinPoint.scala

```diff
@@ -29,14 +29,36 @@ object JoinPoint {
   def mux(cond: Code[Boolean], j1: JoinPoint[Unit], j2: JoinPoint[Unit]): Code[Ctrl] =
     mux((), cond, j1, j2)
 
+  def switch(
+    target: Code[Int],
+    dflt: JoinPoint[Unit],
+    cases: Seq[(Int, JoinPoint[Unit])]
+  ): Code[Ctrl] = {
+    ensureStackIndicator(dflt.stackIndicator)
+    cases.foreach { case (_, j) => ensureStackIndicator(j.stackIndicator) }
+    new Code[Ctrl] {
+      def emit(il: Growable[AbstractInsnNode]): Unit = {
+        target.emit(il)
+        il += new LookupSwitchInsnNode(dflt.label,
```

Comment:
I haven't looked at where you're using this yet. You think lookupswitch is better for your use case than tableswitch?

Reply:
in hindsight, tableswitch is probably better

Reply:
changed to tableswitch / `Seq[JoinPoint[Unit]]` instead of `Seq[(Int, JoinPoint[Unit])]`